### DISCUSSION TITLE
[Enhancement] [dropdown] Dropdown 下拉菜单 希望加一个手动定位，鼠标右键菜单使用

### DISCUSSION THREAD

**Discussion by iZaiZaiA** [element-plus/element-plus#9264] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `dropdown`

## Description

参考 `https://www.naiveui.com/zh-CN/light/components/dropdown#manual-position.vue`

或 `https://arco.design/react/components/dropdown#%E5%8F%B3%E9%94%AE%E8%8F%9C%E5%8D%95`

我希望的是，不需要包裹元素。我自行通过元素的 `@contextmenu` 事件获取 X、Y的坐标，然后，传递给 Dropdown 组件，以此来实现 `鼠标右键菜单` 功能。

因为包裹的话，会渲染太多了，我是用在 `tree` 上，如果有N多个节点，就会渲染很多个 `Dropdown` 没必要，我想只渲染一个 `Dropdown` 然后，通过节点事件来关联，实现鼠标右键菜单。

希望能抽空考虑下这个需求

### Additional comments

_(empty)_

<img width="477" alt="image" src="https://user-images.githubusercontent.com/26807649/183807128-2e29d566-3da3-4694-a1a8-18ca8b6bb927.png">

<img width="417" alt="image" src="https://user-images.githubusercontent.com/26807649/183807142-2f5bbe60-11a8-40c0-b8a1-7f861fee4b7a.png">

**Comment by rzzf** [comment]
has #22056
